### PR TITLE
chefstyle Style/AndOr fixes

### PR DESCRIPTION
**Pull request by lamont-granquist**
this is part of our informal style guide, lets make it formal since
clearly its not getting followed very well.

### CODE REVIEW

**Comment by thommay** [comment]
maximal :+1:

**Comment by lamont-granquist** [comment]
@chef/client-core anyone have any thoughts or objections?  @danielsdeleo ?

File: lib/chef/application/client.rb [hunk 1]
```diff
@@ -480,7 +480,7 @@ def interval_sleep(sec)
   end
 
   def client_sleep(sec)
-    IO.select([ SELF_PIPE[0] ], nil, nil, sec) or return
```
Comment (danielsdeleo) [inline lib/chef/application/client.rb]: This is actually the correct idiomatic usage of `or`, for flow control instead of joining multiple conditions.

**Comment by danielsdeleo** [comment]
A fair number of the things that got "fixed" were actually the correct usage (as a kind of postfix if statement), though lots of them were wrong. I don't know that you can actually enforce correct usage in an automated way. Though if the argument is that the correct usage of and/or is too subtle and we want to swear if off entirely, I suppose I could be convinced. Though in that case it would be nice to use a control flow statement, like `return unless IO.select(*args)` in the example I commented on above.

**Comment by lamont-granquist** [comment]
i think i always use `return unless ...` -- at the beginning of the line i'm more likely to 'see' the return and its easier to scan down the 'inside' of the code to find early returns than it is to scan down the right margin.

**Comment by lamont-granquist** [comment]
@danielsdeleo @thommay can you both look over d4579addae1dafa6b9e39b286995b97dac450aa2 ?

rubocop wasn't smart enough to autofix that, and i'm not sure i'm smart enough to fix the yum change correctly.

and i think that's an excellent example of why i'd prefer to ban AndOr entirely because i really don't want to have to try to decipher that.

**Comment by lamont-granquist** [comment]
it looks like we only used `|| return` that one time in the code here.

File: lib/chef/chef_fs/config.rb [hunk 2]
```diff
@@ -270,7 +270,7 @@ def object_paths
           end
           object_names.each do |object_name|
             # cookbooks -> cookbook_path
-            singular_name = INFLECTIONS[object_name] or raise "Unknown object name #{object_name}"
```
Comment (danielsdeleo) [inline lib/chef/chef_fs/config.rb]: This one also needs a manual fix-up since it was a control flow usage

File: lib/chef/event_loggers/base.rb [hunk 3]
```diff
@@ -42,8 +42,8 @@ def self.available_event_loggers
     end
 
     def self.new(name)
-      event_logger_class = by_name(name.to_s) or
-        raise UnknownEventLogger, "No event logger found for #{name} (available: #{available_event_loggers.join(', ')})"
+      (event_logger_class = by_name(name.to_s)) ||
+        raise(UnknownEventLogger, "No event logger found for #{name} (available: #{available_event_loggers.join(', ')})")
```
Comment (danielsdeleo) [inline lib/chef/event_loggers/base.rb]: `raise unless`

File: lib/chef/formatters/base.rb [hunk 4]
```diff
@@ -51,8 +51,8 @@ def self.available_formatters
     #--
     # TODO: is it too clever to be defining new() on a module like this?
     def self.new(name, out, err)
-      formatter_class = by_name(name.to_s) or
-        raise UnknownFormatter, "No output formatter found for #{name} (available: #{available_formatters.join(', ')})"
+      (formatter_class = by_name(name.to_s)) ||
```
Comment (danielsdeleo) [inline lib/chef/formatters/base.rb]: raise unless

File: lib/chef/http/socketless_chef_zero_client.rb [hunk 5]
```diff
@@ -177,7 +177,7 @@ def req_to_rack(method, url, body, headers)
 
       def to_net_http(code, headers, chunked_body)
         body = chunked_body.join("")
-        msg = STATUS_MESSAGE[code] or raise "Cannot determine HTTP status message for code #{code}"
+        (msg = STATUS_MESSAGE[code]) || raise("Cannot determine HTTP status message for code #{code}")
```
Comment (danielsdeleo) [inline lib/chef/http/socketless_chef_zero_client.rb]: raise unless

File: lib/chef/policy_builder/policyfile.rb [hunk 6]
```diff
@@ -256,10 +256,10 @@ def cookbook_lock_for(cookbook_name)
       # @api private
       def run_list
         if named_run_list_requested?
-          named_run_list or
-            raise ConfigurationError,
+          named_run_list ||
+            raise(ConfigurationError,
```
Comment (danielsdeleo) [inline lib/chef/policy_builder/policyfile.rb]: This whole file should be changed to `raise unless`

**Comment by danielsdeleo** [comment]
https://github.com/chef/chef/commit/d4579addae1dafa6b9e39b286995b97dac450aa2 looks correct to me

**Comment by lamont-granquist** [comment]
this is better, i had to leave some `@foo || raise("stuff")` style code because it was in accessors.  i kinda lean towards wanting parens anyway...

**Pull request merged**